### PR TITLE
16700 Fixed limited restoration expiry months validation and save/resume

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#16700

*Description of changes:*
- app version = 5.11.18
- misc package updates
- renamed some store getters
- updated props and event handlers
- used omit from lodash instead of own package
- added some missing "haveChanges" statements

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: package-lock.json
Comment (severinbeauvais) [inline package-lock.json:1]: This is a large diff because I rebuilt this from scratch.

File: src/components/Restoration/RestorationType.vue [hunk 1]
```diff
@@ -130,15 +127,40 @@ export default class RestorationType extends Mixins(DateMixin, CommonMixin) {
...
+    }
+    return 24 // default if no expiry date was set
+  }
+
+  /**
+   * When months has changed, sets the limited restoration extension expiry date in the store.
+   * @param months The number of months to extend the restoration (or null).
+   */
+  onMonthsChanged (months: number): void {
+    if (Number.isInteger(months)) {
```
Comment (severinbeauvais) [inline src/components/Restoration/RestorationType.vue:144]: This is true if `months` is a number (including 0).

File: src/store/store.ts [hunk 2]
```diff
@@ -1433,8 +1433,13 @@ export const useStore = defineStore('store', {
     setWindowWidth (width: number) {
       this.stateModel.windowWidth = width
     },
+
+    //
+    // alamgamation-specific setters
+    //
     setAmalgamatingBusinesses (amalgamatingBusinesses: Array<AmalgamatingBusinessIF>) {
       this.stateModel.amalgamation.amalgamatingBusinesses = amalgamatingBusinesses
+      if (!this.stateModel.ignoreChanges) this.stateModel.haveChanges = true
```
Comment (severinbeauvais) [inline src/store/store.ts:1442]: "If the user has changed something then set a flag so we can try to stop users from exiting without saving their changes."

There were quite a few of these missing. I only updated some amalgamation and restoration setters.

File: src/store/store.ts [hunk 3]
```diff
@@ -1451,45 +1456,55 @@ export const useStore = defineStore('store', {
...
-    },
-    setBusinessNameValid (valid: boolean) {
-      this.stateModel.restoration.businessNameValid = valid
-    },
-    setRestorationTypeValid (valid: boolean) {
-      this.stateModel.restoration.restorationTypeValid = valid
+
+    //
+    // restoration-specific setters
+    //
```
Comment (severinbeauvais) [inline src/store/store.ts:1471]: I also sorted the setters below.

File: package-lock.json
Comment (ketaki-deodhar) [reply package-lock.json:1]: Just curiosity: why from scratch :)

File: package.json [hunk 4]
```diff
@@ -34,7 +34,7 @@
     "@bcrs-shared-components/help-business-number": "1.1.4",
     "@bcrs-shared-components/interfaces": "1.1.15",
     "@bcrs-shared-components/jurisdiction": "1.1.4",
-    "@bcrs-shared-components/limited-restoration-panel": "1.0.5",
+    "@bcrs-shared-components/limited-restoration-panel": "2.1.0",
```
Comment (JazzarKarim) [inline package.json:37]: Weren't we reserving major patches for the Vue 3 versions of the components Sev?

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-731-i4m8lh6d.web.app

File: package-lock.json
Comment (severinbeauvais) [reply package-lock.json:1]: To update all the dependencies and sub-dependencies.

Consider that the package-lock.json file specifies the exact version of all packages that are imported, which get stale after a while. If I delete it and do `npm i` again, the latest version of everything is imported. I've been trying to solve some npm warnings and unit test warnings and IDE warnings...

File: src/store/store.ts [hunk 5]
```diff
@@ -1451,45 +1456,55 @@ export const useStore = defineStore('store', {
...
+      if (!this.stateModel.ignoreChanges) this.stateModel.haveChanges = true
     },
     setRestorationApprovalType (type: ApprovalTypes) {
       this.stateModel.restoration.approvalType = type
+      if (!this.stateModel.ignoreChanges) this.stateModel.haveChanges = true
+    },
+    setRestorationApprovalTypeValid (valid: boolean) {
+      this.stateModel.restoration.approvalTypeValid = valid
     },
-    setRestorationCourtOrderNumber (fileNumber: string) {
```
Comment (JazzarKarim) [inline src/store/store.ts:1479]: This is obsolete? We don't need it anymore?

**Review by JazzarKarim — APPROVED** [review]

File: src/store/store.ts (hunk 5 above)
Comment (severinbeauvais) [reply src/store/store.ts:1479]: Wasn't used anywhere.

There's another setter to save the court order object, which is used.

Comment (JazzarKarim) [reply src/store/store.ts:1479]: Thanks, resolving.

File: package.json (hunk 4 above)
Comment (severinbeauvais) [reply package.json:37]: Yes. For this component it's 3.x.x.

What you see here is an import of a Vue2 refactor of this component (done for Edit UI) that I pulled in, plus some small changes you saw in that earlier PR.

**Review by ketaki-deodhar — APPROVED** [review]

File: package-lock.json
Comment (severinbeauvais) [reply package-lock.json:1]: The value of the package lock file is that the CI/CD jobs can build the app using the exact same versions that we build locally. We wouldn't want the jobs to always use the latest versions, as there could be changes we need to review.

**Pull request merged**